inc/graphql.php

```diff
@@ -0,0 +1,143 @@
+<?php
+/**
+ * This file contains functionality needed to integrate with WPGraphQL.
+ *
+ * @package Byline_Manager
+ */
+
+namespace Byline_Manager;
+
+use WPGraphQL;
+
+add_action( 'graphql_register_types', __NAMESPACE__ . '\register_byline_types' );
+
+/**
+ * Register data types and field for supporting bylines on posts.
+ */
+function register_byline_types() {
+	// Register a text profile.
+	register_graphql_object_type(
+		'TextProfile',
+		[
+			'description' => __( 'A TextProfile type', 'byline-manager' ),
+			'fields'      => [
+				'text' => [
+					'type'        => 'String',
+					'description' => __( 'Text for a text profile', 'byline-manager' ),
+				],
+			],
+		]
+	);
+
+	// Register a union for both Profile and TextProfile types.
+	register_graphql_union_type(
+		'ProfileTypes',
+		[
+			'typeNames'   => [ 'Profile', 'TextProfile' ],
+			'resolveType' => function ( $profile ) {
+				if (
+					$profile instanceof WPGraphQL\Model\Post
+					&& PROFILE_POST_TYPE === $profile->post_type
+				) {
+					return 'Profile';
+				}
+
+				return 'TextProfile';
+			},
+		]
+	);
+
+	// Register options for byline format.
+	register_graphql_enum_type(
+		'BylineTextEnum',
+		[
+			'description' => __( 'Byline format', 'byline-manager' ),
+			'values'      => [
+				'TEXT'       => [
+					'value' => 'text',
+				],
+				'LINKS'      => [
+					'value' => 'links',
+				],
+				'POST_LINKS' => [
+					'value' => 'post_links',
+				],
+			],
+		]
+	);
+
+	register_graphql_object_type(
```

Comment:
Since the rest of the `register_` calls have a comment, this one told me it feels left out

Reply:
I feel the same way.